plotly/tools.py

```diff
@@ -2978,52 +3692,25 @@ def _unlabel_rgb(colors):
         such colors and returns the color tuples in tuple(s) (a, b, c)
 
         """
-        if isinstance(colors, str):
-            str_vals = ''
-            for index in range(len(colors)):
-                try:
-                    float(colors[index])
+        str_vals = ''
+        for index in range(len(colors)):
+            try:
+                float(colors[index])
+                str_vals = str_vals + colors[index]
+            except ValueError:
+                if (colors[index] == ',') or (colors[index] == '.'):
```

Comment:
🐄 Please Excuse Me Dear Aunt Sally, but no need for the `()` in there. Check out Python's [Order of Operations](http://www.mathcs.emory.edu/~valerie/courses/fall10/155/resources/op_precedence.html). `or` is only evaluated before `lambda`.